microservices/services/query-metric/api/src/main/java/datawave/microservice/querymetric/BaseQueryMetric.java

```diff
@@ -699,6 +702,46 @@ public enum Lifecycle {
         NONE, DEFINED, INITIALIZED, RESULTS, CLOSED, CANCELLED, MAXRESULTS, NEXTTIMEOUT, TIMEOUT, SHUTDOWN, MAXWORK
     }
     
+    public void addSubPlan(String plan, RangeCounts rangeCounts) {
+        synchronized (this.subPlans) {
+            if (subPlans.containsKey(plan)) {
+                RangeCounts combinedCounts = new RangeCounts();
+                RangeCounts currentCounts = subPlans.get(plan);
+                combinedCounts.setDocumentRangeCount(currentCounts.getDocumentRangeCount() + rangeCounts.getDocumentRangeCount());
+                combinedCounts.setShardRangeCount(currentCounts.getShardRangeCount() + rangeCounts.getShardRangeCount());
+                subPlans.put(plan, combinedCounts);
+            } else {
+                subPlans.put(plan, rangeCounts);
+            }
+        }
+    }
+    
+    public void addSubPlans(Map<String,RangeCounts> subplanMap) {
+        synchronized (this.subPlans) {
+            if (subplanMap != null && !subplanMap.isEmpty()) {
+                for (Map.Entry<String,RangeCounts> entry : subplanMap.entrySet()) {
+                    if (subPlans.containsKey(entry.getKey())) {
+                        RangeCounts combinedCounts = new RangeCounts();
+                        RangeCounts currentCounts = subPlans.get(entry.getKey());
+                        combinedCounts.setDocumentRangeCount(currentCounts.getDocumentRangeCount() + entry.getValue().getDocumentRangeCount());
+                        combinedCounts.setShardRangeCount(currentCounts.getShardRangeCount() + entry.getValue().getShardRangeCount());
+                        subPlans.put(entry.getKey(), combinedCounts);
+                    } else {
+                        subPlans.put(entry.getKey(), entry.getValue());
+                    }
+                }
+            }
+        }
+    }
+    
+    public Map<String,RangeCounts> getSubPlans() {
+        return subPlans;
+    }
+    
+    public void setSubPlans(Map<String,RangeCounts> subPlans) {
+        this.subPlans = subPlans;
+    }
+    
```

Comment:
Is the synchronized in addSubplan / addSubplans really necessary?  If so, by returning the object in getSubplans, you are likely allowing non-synchronized access.  Also, if multiple threads are accessing this calss, then setSubplans runs the risk of replacing the object that you have synchronized on.  Or.... maybe you don't really have multile threads accessing a metric at the same time?